docs/api/hardhat/hardhat-zksync-upgradable.md

```diff
@@ -0,0 +1,433 @@
+# `hardhat-zksync-upgradable`
+
+The hardhat-zksync-upgradable plugin is a Hardhat plugin that supports end-to-end pipelines for deploying and updating upgradable smart contracts on the zkSync Era network.
+
+The plugin is based on [@openzeppelin/hardhat-upgrades](https://www.npmjs.com/package/@openzeppelin/hardhat-upgrades) and [@openzeppelin/upgrades-core](https://www.npmjs.com/package/@openzeppelin/upgrades-core), plugins for deploying and managing upgradeable smart contracts on Ethereum network. The hardhat-zkSync-upgradable plugin provides an easy-to-use interface for interacting with the OpenZeppelin Upgrades Plugins within a Hardhat environment on zkSync.
+
+
+## Installation
+
+[@matterlabs/hardhat-zksync-upgradable](https://www.npmjs.com/package/@matterlabs/hardhat-zksync-upgradable)
+
+::: code-tabs
+
+@tab:active yarn
+
+```bash
+yarn add -D @matterlabs/hardhat-zksync-upgradable @openzeppelin/contracts-upgradeable @openzeppelin/hardhat-upgrades
+```
+
+@tab npm
+
+```bash
+npm i -D @matterlabs/hardhat-zksync-upgradable
+```
+
+:::
+
+## Configuration
+
+After installing it, add the plugin to your `hardhat.config.ts` file:
+
+
+```typescript
+
+import '@matterlabs/hardhat-zksync-solc';
+import '@matterlabs/hardhat-zksync-deploy';
+import '@matterlabs/hardhat-zksync-upgradable';
+
+import { HardhatUserConfig } from 'hardhat/config';
+
+const config: HardhatUserConfig = {
+    zksolc: {
+        version: '1.3.7',
+        compilerSource: 'binary',
+        settings: {
+            optimizer: {
+                enabled: true,
+            },
+        },
+    },
+    defaultNetwork: 'zkSyncNetwork',
+    networks: {
+        goerli: {
+            zksync: false,
+            url: 'http://localhost:3050',
+        },
+        zkSyncNetwork: {
+            zksync: true,
+            ethNetwork: 'goerli',
+            url: 'http://localhost:8545',
+        },
+    },
+    solidity: {
+        version: '0.8.19',
```

Comment:
I guess we need to test this with latest Mac M1, and bring it down if it fails.

Reply:
it should be ok now 😉

Reply:
🙏